HR-Data-Supported-Healthcare/Corpora-that-Support-Clinical-Reasoning
# added checks for empty src and dest files. program only loads docx files

## Description
- Added function 'init' which checks whether the src and dest folders exist and creates them if it doesn't
- Added condition in BaseETL which checks whether the Extract has returned any data. Program exits if no data has been extracted.
- Added condition in BaseETL extract to only extract .docx files from the src directory

## Tests (in order)
### Test 1
1. remove ./data and ./output from directory
2. run 'py gen_corpora.py'
**Result: Program should print that source and destination folders have been created and the folders should appear in the directory. Program should exit because there is no data in the src folder**

### Test 2
1. Ensure ./data and ./output exist but are empty
2. run 'py gen_corpora.py'
**Result: Program exits and prints that there are no files in the src folder**

### Test 3
1. Add a random file to ./data such as 'blabla.txt'
2. run 'py gen_corpora.py'
**Result: Program exits and prints that there are no files in the src folder. No exceptions should occur**

### Test 4
1. Add a case docx file to ./data
2. run 'py gen_corpora.py'
**Result: ./output should contain a corpus file**
